src/extremeweatherbench/derived.py

```diff
@@ -0,0 +1,48 @@
+"""
+This module contains functions for calculating derived variables for event types.
+"""
+
+import abc
+import logging
+
+import numpy as np
+import xarray as xr
+
+np.set_printoptions(suppress=True)
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.INFO)
+
+
+class DerivedVariable(abc.ABC):
+    """A class to hold a derived variable for an event type.
+
+    Derived variables are not variables that exist in the original dataset, but can be
+    calculated from the original variables. Certain event types may require derived
+    outputs such as the Craven-Brooks Significant Severe parameter, a variable that is
+    calculated based on temperature, dewpoint, pressure coordinates, and bulk
+    0-6km wind shear.
+
+    Attributes:
+        name: The name of the derived variable.
+
+    Methods:
+        calculate: Calculate the derived variable for the given data.
+    """
+
+    def __init__(self):
+        pass
+
+    @property
+    def name(self) -> str:
+        """Return the class name without parentheses."""
+        return self.__class__.__name__
+
+    @abc.abstractmethod
+    def calculate(self, data: xr.Dataset | xr.DataArray) -> xr.DataArray:
```

Comment:
I think this is too flexible of a signature; I'd propose that we restrict this to an `xr.Dataset`, and require that all the fields necessary to compute the derived variable must be contained in that structure.

Another thing to consider - we likely need at least a `**kwargs` in the abstract signature, since some methods for deriving variables may have specific parameters that need customization.